### ISSUE TITLE
Add tests for new quest structs and implement corresponding functions

### ISSUE THREAD

**Issue by ddelgado60** [uchicago-cs/chiventure#1016]
Currently, we are waiting on a pull request for the new switch from a linked list data structure to a non-binary tree data structure to contain achievements. Once this is done, we have new data structures that allow for passive and active quests, new reward types, and a stats requirement before being able to start a quest. 
In particular, we plan to remodel existing code to account for this, as well as writing new functions for the new data types. We will also write new tests for the new functions as well.

assignees: Devon, Gabi

**Comment by gabigarc03** [comment]
Tests have now been rewritten for the new data structures and merged into dev. See PR #1096 for more details.
